File: Week 1/Day 1/algo-linear-search/code.js
const arrayToSearchThrough = [];
for (let i = 1; i <= 1000; i++) {
  arrayToSearchThrough.push(i);
}

exports.linearSearch = function (valueToFind, arrayToSearchThrough) {
  for (let i = 0; i < arrayToSearchThrough.length; i++) {
    if (arrayToSearchThrough[i] === valueToFind) {
      return i;
    }
  }
};

exports.linearSearchGlobal = function (valueToFind, arrayToSearchThrough) {
  let indexes = [];
  for (let i = 0; i < arrayToSearchThrough.length; i++) {
    if (arrayToSearchThrough[i] === valueToFind) {
      indexes.push(i);
    }
  }
  return indexes;
};

File: Week 1/algo-linear-search/code.py
array_to_search_through = []
for number in range(1, 1001):
    array_to_search_through.append(number)


def linear_search(value_to_find, array_to_search_through):
    for i in range(len(array_to_search_through)):
        if (array_to_search_through[i] == value_to_find):
            return i


def linear_search_global(value_to_find, array_to_search_through):
    arr = []
    for i in range(len(array_to_search_through)):
        if (array_to_search_through[i] == value_to_find):
            arr.append(i)
    return arr
